### PR TITLE
*: create int primary key table as nonclustered by default

### PR DESCRIPTION
**Pull request by lysu**
Signed-off-by: lysu <sulifx@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

- [x] tidb-test https://github.com/pingcap/tidb-test/pull/1167
- [x] update BR CI https://github.com/pingcap/br/pull/830
- [ ] update CDC CI https://github.com/pingcap/ticdc/pull/1503
- [ ] update dumpling CI [manually test]
- [ ] update tics 
- [x] update tispark https://github.com/pingcap/tispark/pull/1938

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed, How it Works:

- create nonclustered primary key for int  as default
- make test case pass(so add a invisible variable --- `@@tidb_int_primary_key_default_as_clustered`  to help change test)

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`: In clustered index project - TODO

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - default Int PK table will have addition lookup from index to table data.
    - but hotspot problem can be better handle in default settings

### Release note 

- No release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/23046)

### CODE REVIEW

**Comment by overvenus** [comment]
This PR fails br tests https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/5558/pipeline/73#step-1062-log-45

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1447,9 +1447,14 @@ func buildTableInfo(
 					}
 				}
 			case model.PrimaryKeyTypeDefault:
+				// AlterPrimaryKey = true ----> all pk must be nonclustered
+				// AlterPrimaryKey = false + EnableClusteredIndex = true + noBinlog ---> all pk must be clustered
```
Comment (wjhuang2016) [inline ddl/ddl_api.go:1451]: (AlterPrimaryKey = false) + (EnableClusteredIndex = true) + noBinlog ---> all pk must be clustered

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -818,6 +818,10 @@ type SessionVars struct {
 	// AllowFallbackToTiKV indicates the engine types whose unavailability triggers fallback to TiKV.
 	// Now we only support TiFlash.
 	AllowFallbackToTiKV map[kv.StoreType]struct{}
+
+	// IntPrimaryKeyDefaultAsClustered indicates whether create integer primary table as clustered
+	// when `@@tidb_enable_clustered_index=0` and create-table stmt without `clustered`, like 4.0 behavior.
```
Comment (wjhuang2016) [inline sessionctx/variable/session.go:823]: s/like 4.0 behavior/ If it's true, the behavior is the same as the TiDB 4.0 and the below versions.

**Pull request merged**